### PR TITLE
[tests-only][full-ci]Extend tests scenario for space notification

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds tests for notification
- Scenario:  get a notification of space shared and disabled
- Scenario: get a notification of space unshared
- Scenario: all notification related to space gets deleted when sharer delete that resource
- Scenario: sharee doesn't get any notification after removing user from space

## Related Issue

- https://github.com/owncloud/ocis/issues/6060

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiNotification/spaceNotification.feature [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+  Scenario: user doesn't get any notification after removing user from space
+    Given user "Alice" has shared a space "notification checking" with settings:
+      | shareWith | Brian  |
+      | role      | editor |
+    And user "Alice" has unshared a space "notification checking" shared with "Brian"
+    And user "Alice" has disabled a space "notification checking"
+    When user "Brian" lists all notifications
+    Then the HTTP status code should be "200"
+    And user "Brian" should have "2" notifications
+    And the JSON response should not contain a notification message with the subject "Space disabled" and the message-details should match
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/spaceNotification.feature:473]: JSON assertion realy helpful in negation cases?

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 2]
```diff
@@ -2188,6 +2210,29 @@ public function sendRemoveObjectFromSpaceRequest(
 		);
 	}
 
+	/**
+	 * @Given /^user "([^"]*)" has deleted a space "([^"]*)"$/
+	 * @Given /^user "([^"]*)" has deleted a space "([^"]*)" owned by user "([^"]*)"$/
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2215]: I didn't see its usage
```suggestion
```

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/22862/89/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/22862/tracing/members-of-the-space-can-control-the-versions-of-the-files-alice-2023-6-1-11-35-34.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/22862/tracing/public-link-for-space-alice-2023-6-1-11-36-10.zip 


</p></details>

File: tests/acceptance/features/apiNotification/notification.feature [hunk 3]
```diff
@@ -1,7 +1,7 @@
 @api
 Feature: Notification
   As a user
-  I want to be notified of various events
+  I want to be notified of actions related to me
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/notification.feature:4]: I would prefer the previous one more

File: tests/acceptance/features/bootstrap/NotificationContext.php [hunk 4]
```diff
@@ -115,6 +115,28 @@ public function theNotificationsShouldBeEmpty(): void {
...
+	/**
+	 * @Then /^user "([^"]*)" should have "([^"]*)" notifications$/
+	 *
+	 * @param string $user
+	 * @param int $numberOfNotification
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function userShouldHaveNotifications(string $user, int $numberOfNotification) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/NotificationContext.php:127]:
```suggestion
	public function userShouldHaveNotifications(string $user, int $numberOfNotification): void {
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 5]
```diff
@@ -2187,6 +2209,28 @@ public function sendRemoveObjectFromSpaceRequest(
...
+	public function userHasDeletedASpaceOwnedByUser(
+		string $user,
+		string $spaceName,
+		string $owner = ''
+	):void {
+		$this->sendDeleteSpaceRequest($user, $spaceName, $owner);
+		$this->featureContext->theHTTPStatusCodeShouldBe(
+			204,
+			"Expected response status code should be 200"
+		);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2231]:
```suggestion
	 * @param  string $user
	 * @param  string $spaceName
	 *
	 * @return void
	 * @throws GuzzleException
	 */
	public function userHasDeletedASpaceOwnedByUser(
		string $user,
		string $spaceName
	):void {
		$this->sendDeleteSpaceRequest($user, $spaceName);
		$this->featureContext->theHTTPStatusCodeShouldBe(
			204,
			"Expected response status code should be 200"
		);
```

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 6]
```diff
@@ -2132,6 +2132,28 @@ public function userHasSharedSpace(
 		$this->ocsContext->theOCSStatusCodeShouldBe($expectedOCSStatus, "Expected OCS response status code $expectedOCSStatus");
 	}
 
+	/**
+	 * @Given user :arg1 has unshared a space :arg2 shared with :arg3
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SpacesContext.php:2136]:
```suggestion
	 * @Given user :user has unshared a space :spaceName shared with :recipient
```

File: tests/acceptance/features/apiNotification/notification.feature [hunk 7]
```diff
@@ -265,21 +444,29 @@ Feature: Notification
...
+      | de       | Space freigegeben | Alice Hansen hat Sie zu Space notification checking hinzugefügt |
+      | es       | Space compartido  | Alice Hansen te añadió al Space notification checking           |
 
 
-  Scenario Outline: notifications related to a resource get deleted when the resource is deleted
-    Given user "Alice" has shared entry "<resource>" with user "Brian"
-    And user "Brian" has accepted share "/<resource>" offered by user "Alice"
-    And user "Alice" has unshared entity "<resource>" shared to "Brian"
-    And user "Alice" has deleted entity "/<resource>"
+  Scenario: all notification related to space gets deleted when sharer delete that resource
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/notification.feature:452]:
```suggestion
  Scenario: all notifications related to space get deleted when the sharer deletes that space
```

File: tests/acceptance/features/apiNotification/notification.feature [hunk 8]
```diff
@@ -265,21 +444,29 @@ Feature: Notification
...
     When user "Brian" lists all notifications
     Then the HTTP status code should be "200"
     And the notifications should be empty
-    Examples:
-      | resource      |
-      | textfile1.txt |
-      | my_data       |
+
+
+  Scenario: user doesn't get any notification after removing user from space
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/notification.feature:464]:
```suggestion
  Scenario: user doesn't get any notification after being removed from space
```

File: tests/acceptance/features/apiNotification/notification.feature [hunk 9]
```diff
@@ -10,241 +10,420 @@ Feature: Notification
...
+    And the administrator has given "Alice" the role "Space Admin" using the settings api
+    And user "Alice" has created a space "notification checking" with the default quota using the GraphApi
 
 
-  Scenario Outline: user gets a notification of resource sharing
-    Given user "Alice" has shared entry "<resource>" with user "Brian"
+  Scenario: get a notification of space shared
+    Given user "Alice" has shared a space "notification checking" with settings:
+      | shareWith | Brian  |
+      | role      | editor |
```
Comment (saw-jan) [inline tests/acceptance/features/apiNotification/notification.feature:20]: Shouldn't this file be for notifications related to resources?

**Review by saw-jan — CHANGES_REQUESTED** [review]
Please, add only the necessary scenarios and then we can do the moving part in other PRs.
Scenarios related to resources got disappeared.

File: tests/acceptance/features/apiNotification/notification.feature (hunk 9 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiNotification/notification.feature]: i messed up while git push 🥲

**Comment by amrita-shrestha** [comment]
> Please, add only the necessary scenarios and then we can do the moving part in other PRs. Scenarios related to resources got disappeared.

@saw-jan I don't think this PR contains huge changes to break down into multiple PR.

**Comment by saw-jan** [comment]
> > Please, add only the necessary scenarios and then we can do the moving part in other PRs. Scenarios related to resources got disappeared.
> 
> @saw-jan I don't think this PR contains huge changes to break down into multiple PR.
Yeah but it would be easier for you if you don't do the file move here. You can add the new scenarios and do the moving in next PR. This way, there will be less chance of scenarios being lost.

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

**Review by saw-jan — CHANGES_REQUESTED** [review]
@amrita-shrestha again the same issue occurred - notification tests related to resource disappeared. Please, pull the branch first if you are working from a different machine.

**Review by saw-jan — APPROVED** [review]

**Pull request merged**